src/components/SideBar/index.jsx

```diff
@@ -18,7 +20,19 @@ function SideBar(props) {
   const navigate = useNavigate();
   const [openNewUpload, setOpenNewUpload] = useState(false);
   const user = auth.currentUser;
+  const [openNotificationPopup, setOpenNotificationPopup] = useState(false);
+  const handleOpenNotificationPopup = () => {
+    setOpenNotificationPopup(true);
+  };
+  const [requests, setRequests] = useState([]);
 
+  const acceptRequest = (request) => {
```

Comment:
you don't need to filter the data. Instead here acceptrequest will add the person to the friendlist and rejectrequest wont do such thing. But one thing is common after this, they will update the requests collection. I think db function remove one or delete one shall do. Else you'll manually have to update the data.